### ISSUE TITLE
Mutability compile errors

### ISSUE THREAD

**Issue by palfrey** [mheese/rust-pkcs11#18]
Running `cargo test` (both against Rust 1.40.0 and 1.32.0) got me lots of errors mostly of this form:

```rust
   --> src/tests.rs:734:49
    |
734 |       let res = ctx.get_attribute_value(sh, oh, &template);
    |                                                 ^^^^^^^^^ types differ in mutability
    |
    = note: expected type `&mut std::vec::Vec<types::CK_ATTRIBUTE>`
               found type `&std::vec::Vec<types::CK_ATTRIBUTE>`
```

**Comment by ramtej** [comment]
Yes, I had the same issue as well. Solved with: 

  let mechanism = CK_MECHANISM {
    mechanism: CKM_AES_KEY_GEN,
    pParameter: ptr::**null_mut()**,
    ulParameterLen: 0,
  };

**Comment by palfrey** [comment]
This should now be fixed now #19 is merged
